src/include/sof/llext_manager.h

```diff
@@ -11,10 +11,16 @@
 struct comp_driver;
 struct comp_ipc_config;
 
+#if CONFIG_LLEXT
 uintptr_t llext_manager_allocate_module(struct processing_module *proc,
 					const struct comp_ipc_config *ipc_config,
 					const void *ipc_specific_config);
 
 int llext_manager_free_module(const uint32_t component_id);
+#else
+#define llext_manager_allocate_module(proc, ipc_config, ipc_specific_config) 0
+#define llext_manager_free_module(component_id) 0
+#define llext_unload(ext) 0
```

Comment:
Wow, this looks really weird.... I think it's not just "creative", unusual and harder to read...

- I don't think having these being either functions or macros depending on Kconfig is a good idea. Debugging Kconfig and preprocessing issues is painful enough already, so having totally different error messages depending on CONFIG_LLEXT seems wrong. Please keep these functions always: `static inline uintptr_tllext_manager_free_module(component_id) { return 0; }`. The generated code should be the same.

- Naive question sorry: why are these functions needed when CONFIG_LLEXT is false? I don't know all the details sorry but returning NULLs and zeros is an anti-pattern. Ideally, these shouldn't even be needed so a short comment/pointer explaining why would not hurt.

- where is `llext_unload()` defined when CONFIG_LLEXT is true?

Reply:
> * Naive question sorry: why are these functions needed when CONFIG_LLEXT is false? I don't know all the details sorry but returning NULLs and zeros is an anti-pattern. Ideally, these shouldn't even be needed so a short comment/pointer explaining why would not hurt.

@marc-hb because they're called from lib_manager.c and that one is included in TGL builds too. No, I don't think it's needed there for the same reason as LLEXT - no memory mapping support.

> 
> * where is `llext_unload()` defined when CONFIG_LLEXT is true?

in Zephyr